argument 1 value ‘18446744073709551611’ exceeds maximum object size 9223372036854775807

I just cloned the latest version and built the toolchain from source build-toolchain.sh in the tools folder.  This ran correctly. 
The N64_INST var is set to ~/libdragon.
 
When running build.sh from the tools folder i get the error:

`````
dumpdfs/dumpdfs.c: In function ‘main’:
dumpdfs/dumpdfs.c:801:29: error: argument 1 value ‘18446744073709551611’ exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
  801 |             uint8_t *data = malloc( (size_t)dfs_size( fl ) );
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from dumpdfs/dumpdfs.c:4:
/usr/include/stdlib.h:539:14: note: in a call to allocation function ‘malloc’ declared here
  539 | extern void *malloc (size_t __size) __THROW __attribute_malloc__
      |              ^~~~~~
dumpdfs/dumpdfs.c:837:29: error: argument 1 value ‘18446744073709551611’ exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
  837 |             uint8_t *data = malloc( (size_t)dfs_size( fl ) );
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from dumpdfs/dumpdfs.c:4:
/usr/include/stdlib.h:539:14: note: in a call to allocation function ‘malloc’ declared here
  539 | extern void *malloc (size_t __size) __THROW __attribute_malloc__
      |              ^~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:11: dumpdfs/dumpdfs.o] Error 1
make[1]: *** Waiting for unfinished jobs....
``````
I'm using the latest version of WSL. 

The older version of libdragon which i built from source just a couple of weeks ago had no issue running build.sh

--- a/tools/dumpdfs/dumpdfs.c
+++ b/tools/dumpdfs/dumpdfs.c
@@ -1,15 +1,24 @@
+#define _GNU_SOURCE
 #include <stdio.h>
 #include <string.h>
 #include <stdint.h>
 #include <stdlib.h>
+#include <assert.h>
 #include "dragonfs.h"
 #include "dfsinternal.h"
+#include "../common/polyfill.h"
 
 #if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
 #define SWAPLONG(i) (i)
 #else
 #define SWAPLONG(i) (((uint32_t)((i) & 0xFF000000) >> 24) | ((uint32_t)((i) & 0x00FF0000) >>  8) | ((uint32_t)((i) & 0x0000FF00) <<  8) | ((uint32_t)((i) & 0x000000FF) << 24))
 #endif
+
+struct directory_entry root_dirent = {
+    .next_entry = SWAPLONG(ROOT_NEXT_ENTRY),
+    .flags = SWAPLONG(ROOT_FLAGS),
+    .path = ROOT_PATH,
+};
 
 /* Directory walking flags */
 enum
@@ -722,8 +731,8 @@ void usage(void)
 {
     printf("dumpdfs - Dump the contents of a Dragon FS\n\n");
     printf("Usage:\n");
-    printf("   dumpdfs -l file.dfs -- List contents\n");
-    printf("   dumpdfs -e file.dfs file -- Extract single file to stdout\n");
+    printf("   dumpdfs -l <file.dfs|file.z64> -- List contents\n");
+    printf("   dumpdfs -e <file.dfs|file.z64> file -- Extract single file to stdout\n");
 }
 
 int main( int argc, char *argv[] )
@@ -760,7 +769,19 @@ int main( int argc, char *argv[] )
             fread( filesystem, 1, lSize, fp );
             fclose( fp );
 
-            if (dfs_init_pc( filesystem, 1 ) != DFS_ESUCCESS)
+            int offset = 0;
+            if (strstr(argv[2], ".z64"))
+            {
+                void *fs = memmem(filesystem, lSize, &root_dirent, sizeof(root_dirent));
+                if (!fs)
+                {
+                    fprintf(stderr, "cannot find DragonFS in ROM\n");
+                    return -1;
+                }
+                offset = fs - filesystem;
+            }
+
+            if (dfs_init_pc( filesystem+offset, 1 ) != DFS_ESUCCESS)
             {
                 fprintf(stderr, "Invalid DragonFS filesystem\n");
                 return -1;
@@ -791,7 +812,19 @@ int main( int argc, char *argv[] )
             fread( filesystem, 1, lSize, fp );
             fclose( fp );
 
-            if (dfs_init_pc( filesystem, 1 ) != DFS_ESUCCESS)
+            int offset = 0;
+            if (strstr(argv[2], ".z64"))
+            {
+                void *fs = memmem(filesystem, lSize, &root_dirent, sizeof(root_dirent));
+                if (!fs)
+                {
+                    fprintf(stderr, "cannot find DragonFS in ROM\n");
+                    return -1;
+                }
+                offset = fs - filesystem;
+            }
+
+            if (dfs_init_pc( filesystem+offset, 1 ) != DFS_ESUCCESS)
             {
                 fprintf(stderr, "Invalid DragonFS filesystem\n");
                 return -1;

--- a/tools/dumpdfs/dumpdfs.c
+++ b/tools/dumpdfs/dumpdfs.c
@@ -831,10 +831,17 @@ int main( int argc, char *argv[] )
             }
             
             int fl = dfs_open( argv[3] );
-            uint8_t *data = malloc( (size_t)dfs_size( fl ) );
+            if (fl < 0)
+            {
+                fprintf(stderr, "File %s not found\n", argv[3]);
+                return -1;
+            }
+            int sz = dfs_size( fl );
+            assert( sz >= 0 );
+            uint8_t *data = malloc( sz );
 
-            dfs_read( data, 1, dfs_size( fl ), fl );
-            fwrite( data, 1, dfs_size( fl ), stdout );
+            dfs_read( data, 1, sz, fl );
+            fwrite( data, 1, sz, stdout );
             dfs_close( fl );
 
             free( filesystem );
@@ -867,10 +874,17 @@ int main( int argc, char *argv[] )
             dfs_read( &unused, 1, 4, nu );
             
             int fl = dfs_open( argv[3] );
-            uint8_t *data = malloc( (size_t)dfs_size( fl ) );
+            if (fl < 0)
+            {
+                fprintf(stderr, "File %s not found\n", argv[3]);
+                return -1;
+            }
+            int sz = dfs_size( fl );
+            assert( sz >= 0 );
+            uint8_t *data = malloc( sz );
 
-            dfs_read( data, 1, dfs_size( fl ), fl );
-            fwrite( data, 1, dfs_size( fl ), stdout );
+            dfs_read( data, 1, sz, fl );
+            fwrite( data, 1, sz, stdout );
             dfs_close( fl );
             dfs_close( nu );
 

